scenes/game/ai_player.gd

```diff
@@ -206,129 +323,77 @@ class NumberFromRangeAction:
-	game_state.opponent_state.buddy_locations = opponent.buddy_locations
-	game_state.opponent_state.exceed_cost = opponent.get_exceed_cost()
-	game_state.opponent_state.exceeded = opponent.exceeded
-	game_state.opponent_state.reshuffle_remaining = opponent.reshuffle_remaining
-
-	game_state.active_strike.active = _game_logic.active_strike != null
-	if game_state.active_strike.active:
-		game_state.active_strike.initiator = _game_logic.active_strike.initiator.my_id
-		game_state.active_strike.initiator_card_id = -1
-		if _game_logic.active_strike.initiator_card:
-			game_state.active_strike.initiator_card_id = _game_logic.active_strike.initiator_card.id
-		game_state.active_strike.initiator_ex_card_id = -1
-		if _game_logic.active_strike.initiator_ex_card:
-			game_state.active_strike.initiator_ex_card_id = _game_logic.active_strike.initiator_ex_card.id
-		game_state.active_strike.defender_card_id = -1
-		if _game_logic.active_strike.defender_card:
-			game_state.active_strike.defender_card_id = _game_logic.active_strike.defender_card.id
-		game_state.active_strike.defender_ex_card_id = -1
-		if _game_logic.active_strike.defender_ex_card:
-			game_state.active_strike.defender_ex_card_id = _game_logic.active_strike.defender_ex_card.id
-
-
-func take_turn(game_logic : LocalGame, my_id : Enums.PlayerId):
-	var me = game_logic._get_player(my_id)
-	var opponent = game_logic._get_player(game_logic.get_other_player(my_id))
-	# Decide which action makes the most sense to take.
-	var possible_actions = determine_possible_turn_actions(game_logic, me, opponent)
-	update_ai_state(game_logic, me, opponent)
+func take_turn():
+	game_state.update()
+	var possible_actions = determine_possible_turn_actions()
 	return ai_policy.pick_turn_action(possible_actions, game_state)
 
-func take_boost(game_logic : LocalGame, my_id : Enums.PlayerId, valid_zones : Array, limitation : String, ignore_costs : bool, boost_amount : int) -> BoostAction:
-	var me = game_logic._get_player(my_id)
-	var opponent = game_logic._get_player(game_logic.get_other_player(my_id))
-	# Decide which action makes the most sense to take.
-	var possible_actions = get_boost_actions(game_logic, me, opponent, valid_zones, limitation, ignore_costs, boost_amount)
-	update_ai_state(game_logic, me, opponent)
+func take_boost(valid_zones : Array, limitation : String, ignore_costs : bool, boost_amount : int) -> BoostAction:
+	game_state.update()
+	var possible_actions = get_boost_actions(valid_zones, limitation, ignore_costs, boost_amount)
 	return ai_policy.pick_boost_action(possible_actions, game_state)
 
-func determine_possible_turn_actions(game_logic : LocalGame, me : LocalGame.Player, opponent : LocalGame.Player):
+# TODO: Change all references to game_player below to consult game_state
+# instead? And possibly take game_state as an input if we ever need to generate
+# possible actions out of a hypothetical scenario.
+
+func determine_possible_turn_actions():
```

Comment:
my thinking with always passing in everything and not using member variables was to make it easy to track exactly what state the AI logic is referencing and ensuring it isn't using any state outside of what should be available to the AI at that time or changing anything it shouldn't be.

That said, since I was passing in the whole game anyway it's kind of irrelevant.

I'm open to whichever design you think makes the most sense and is easiest to work with.

Reply:
I think you're right; these will all eventually ingest a full game state in the form of AIGameState. I figured I'd build that out in a later PR once there's functionality and unit tests that make use of it.